### ISSUE TITLE
implement patricia merkle trees

### ISSUE THREAD

**Issue by thepiwo** [aeternity/elixir-node#253]
docs (maybe addtitonal research is needed):
 - https://github.com/ethereum/wiki/wiki/Patricia-Tree
 - https://ethereum.stackexchange.com/questions/6415/eli5-how-does-a-merkle-patricia-trie-tree-work
 - https://blog.ethereum.org/2015/11/15/merkling-in-ethereum/

**Comment by thepiwo** [comment]
@meivantodorov @HristianHristov this is stuck for 11 days now, can we PR and merge?

**Comment by meivantodorov** [comment]
@thepiwo this is still work in progress, and it is not `stuck`. When it is ready, there will be PR.

**Comment by thepiwo** [comment]
when do you think it will be ready, as #341 is depended on this and there have not been any commits in 11 days.

**Comment by meivantodorov** [comment]
The work was focused on improving [this](https://github.com/exthereum/merkle_patricia_tree) library. I hope that after 2 days there will be PR.
